backend/src/test/kotlin/com/group7/artshare/controller/UserInteractionControllerTest.kt

```diff
@@ -0,0 +1,89 @@
+package com.group7.artshare.controller
+
+import com.group7.artshare.entity.*
+import com.group7.artshare.service.JwtService
+import com.group7.artshare.service.ProfileService
+import org.junit.jupiter.api.Assertions.*
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.extension.ExtendWith
+import org.mockito.InjectMocks
+import org.mockito.Mock
+import org.mockito.Mockito.anyString
+import org.mockito.Mockito.`when`
+import org.mockito.junit.jupiter.MockitoExtension
+import org.springframework.http.HttpStatus
+import org.springframework.web.server.ResponseStatusException
+import java.util.*
+
+
+@ExtendWith(MockitoExtension::class)
+internal class UserInteractionControllerTest {
+
+    @InjectMocks
+    lateinit var userInteractionController: UserInteractionController
+
+    @Mock
+    lateinit var profileService: ProfileService
+
+    @Mock
+    lateinit var jwtService: JwtService
+
+    @Test
+    fun successfullyFollowsUser() {
+        val accountInfo1 = AccountInfo("email@email.com", "janedoe", "31415926")
+        val mockUser = RegisteredUser(accountInfo1, setOf())
+        `when`(profileService.followUser("username", mockUser)).thenReturn(HttpStatus.ACCEPTED)
+        `when`(jwtService.getUserFromAuthorizationHeader("authorizationHeader")).thenReturn(mockUser)
+        val response = userInteractionController.followUser("username", "authorizationHeader")
+        assert(response == Unit)
+    }
+
+    @Test
+    fun failsIfFollowItself() {
+        val accountInfo1 = AccountInfo("email@email.com", "janedoe", "31415926")
+        val mockUser = RegisteredUser(accountInfo1, setOf())
+        `when`(
+            profileService.followUser(
+                "username",
+                mockUser
+            )
+        ).thenThrow(ResponseStatusException(HttpStatus.BAD_REQUEST, "User cannot follow itself"))
+        `when`(jwtService.getUserFromAuthorizationHeader("authorizationHeader")).thenReturn(mockUser)
+        assertThrows(ResponseStatusException::class.java) {
+            userInteractionController.followUser(
+                anyString(),
+                "someAuthorization"
+            )
+        }
+    }
+
+    @Test
+    fun failsIfCouldNotFindFollowedUser() {
```

Comment:
The same goes here. I think you should mock, written as a pseudocode, iregisteredUserService.findByUsername("username") to give the necessary error implying no user found that is given when we run the program in debug mode and get. Then asserThrows with giving "username" instead of anyString() in line 71.

Reply:
You are right, thanks for pointing out. Changed in the latest commit.